### ISSUE TITLE
Kiali operator helmchart frequently changes the replica count when HPA is enabled

### ISSUE THREAD

**Issue by rkadam-infracloud** [kiali/kiali#7559]
### Describe the bug

When HPA is enabled in Kiali Operator helm chart CR after removing the `replicas` parameter, the number of replica continously bounce between 1 and min_replica count in HPA. 

### Expected Behavior
If HPA is enabled, it should set to min count and then add replica when the threshold is crossed by pods.

### What are the steps to reproduce this bug?

1. Install Kiali operator helm chart, initialy without HPA and replica count as 3
2. Remove replicas parameter and add HPA with min 3 and max 6 replica, any setting for CPU or memory threshold. At this point the CR looks like this
```
cr:
  create: true
  namespace: kiali-operator
  spec:
    deployment:
      hpa:
        spec:
          maxReplicas: 6
          minReplicas: 3
          metrics:
          - type: Resource
            resource:
              name: cpu
              target:
                type: Utilization
                averageUtilization: 70
          - type: Resource
            resource:
              name: memory
              target:
                type: Utilization
                averageUtilization: 70
      resources:
        requests:
          memory: "1Gi"
        limits: 
          memory: "2Gi"
    istio_namespace: istio-system
    auth:
      strategy: anonymous
    external_services:
      prometheus:
        url: "http://prometheus-server.prometheus:80/"
      tracing:
        enabled: false
      grafana:
        enabled: false
```
3. ideally the replica count should match min replica count in HPA. However, what actually happens is the replica count fluctuates continuously between 1 and 3

### Environment
Found this issue on argocd managed setup, but can be replicated with minikube too

- **Kiali version:** v1.83.0
- **Istio version:** 1.21.2
- **Kubernetes impl:**  v1.30.1
- **Kubernetes version:** v1.30.1
- **Other notable environmental factors:** Istio 1.21.2

**Comment by jmazzitelli** [comment]
Can you provide actual yaml examples for your replication procedures so it is as simple as possible for people to replicate? e.g. `add HPA with min 3 and max 6 replica, any setting for CPU or memory threshold` - what does the Kiali CR yaml section look like for this when you do it?

**Comment by rkadam-infracloud** [comment]
I think this is hapenning because the deployment does not considers if HPA is enabled or not, and adds the replica number. This causes race condition between deployment and HPA
https://github.com/kiali/kiali-operator/blob/master/roles/default/kiali-deploy/templates/kubernetes/deployment.yaml#L8
https://github.com/kiali/kiali-operator/blob/v1.87.0/crd-docs/cr/kiali.io_v1alpha1_kiali.yaml#L197

I will add the actual config we are using

**Comment by rkadam-infracloud** [comment]
Suggestion :
we can use a condition like this to set the replica count in deployment
`{% if kiali_vars.deployment.hpa.enabled != true  %}`
so that it takes `replicas` parameter only when HPA is disabled

**Comment by rkadam-infracloud** [comment]
@jmazzitelli i have added the example CR

**Comment by jmazzitelli** [comment]
OK, I can see this happen -- it doesn't bounce, though, unless you touch the Kiali CR. If anyone (or anything, like Argo) modifies the Kiali CR (changing anything in it), then this causes the operator to trigger a reconciliation of the Deployment, resetting the replicas to whatever was in the Kiali CR (default is 1). If you also have HPA, that will then kick in and change it back. So the problem happens when you touch the Kiali CR.

I created two simple PRs for the [server helm chart](https://github.com/kiali/helm-charts/pull/275) and the [operator](https://github.com/kiali/kiali-operator/pull/798) - the replicas will not be explicitly set in the Deployment if HPA is also defined. Test procedures are included in the PRs.

I did not add any additional configuration settings for this - we are really close to the next release (going to be released Monday) and adding a new setting would require more changes and more testing. The PRs to simply fix this bug cause minimal changes and as such I should be able to get this merged in by end of the week so it will make it in this upcoming release.
